.github/workflows/tidy-modern.yml

```
@@ -0,0 +1,26 @@
+name: tidy-modern
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  tidy:
+    runs-on: ubuntu-latest
```

If you choose a docker image of PCL, you don't need to install dependencies (I think you'd need to add clang-tools in the docker image)

---

OK. @shrijitsingh99, is this as simple as adding a `run: docker run ./.dev/docker/env/Dockerfile` to the action before the other commands?

---

No need for that. Actions already provides  a nice interface to pull and run docker.
This is all you need to do.
```suggestion
    runs-on: ubuntu-latest
    container:
      image: pointcloudlibrary/env:18.04
```

---

Nice, I'll give it a try, thank you.

---

@shrijitsingh99 Well, now I get this message:

```
To create a local Git repository instead, add Git 2.18 or higher to the PATH
```
I thought that the checkout action checked out the branch that triggered the action but I guess not.

Is there an action to do that?

---

It does check it out for you, but AFAIK git will not be available to you inside the docker container, you will have to install it manually.

---

By the time I install `git`, Docker has downloaded the repo via REST. Here's what I need to do:
```
The repository will be downloaded using the GitHub REST API
To create a local Git repository instead, add Git 2.18 or higher to the PATH
```
Where do I modify the Docker `PATH`?

---

NVM, I got it sorted, but the 18.04 build is building a different set of files than my 20.04 environment. @kunaltyagi why should that be?